pkgs/development/compilers/ghc/8.10.7.nix

```diff
@@ -151,19 +151,12 @@ let
         "ld.gold" = cc.bintools;
 
         otool = cc.bintools.bintools;
+        install_name_tool = cc.bintools.bintools;
 
-        # GHC needs install_name_tool on all darwin platforms. On aarch64-darwin it is
-        # part of the bintools wrapper (due to codesigning requirements), but not on
-        # x86_64-darwin. We decide based on target platform to have consistent tools
-        # across all GHC stages.
```

Comment:
This comment needs to be rephrased and replace the one for strip below. (In all touched files.) Having the information in a comment is better than piecing it together from a deleted comment and the commit message :)

Reply:
What’s the context of the comment for `strip`? Darwin now uses the default wrapper, which just passes `--enable-deterministic-archives`. Should it not do that with GHC, or should it use the wrapper unconditionally?

```
# TODO(@sternenseemann): also use wrapper if linker == "bfd" or "gold"
```

Reply:
I updated the comments.

Reply:
We want to use the wrapper if possible, *but* unfortunately it is kind of difficult to know when we can use it. I think the `TODO` is about the fact that you can sort of bodge detect GNU binutils using `linker == "gold" || linker == "bfd"` which means there's a wrapper (like on aarch64-darwin and now x86_64-darwin). IIRC there's no wrapper if bintools come from LLVM, so that's a case you'd need to guard against.

I think I never really attempted to improve it since the wrapper is inconsequential for GHC itself and I was dissatisfied with the detectability. I think this mess ought to be cleaned up in a different way, i.e. just symlinking from the wrapper to the unwrapped derivation if an unwrapped bintool is available.

Reply:
I take it using `bintools.isGNU` is unreliable? If not, I can update the GHC files to check `cc.bintools.isGNU || stdenv.targetPlatform.isDarwin`. Otherwise, I’d like to keep the changes just to Darwin to avoid increasing the scope of this PR more.

(Also, please let me know if any more revisions are needed to the comments I added to GHC regarding `install_name_tool` and `strip`.)

Reply:
I don't know whether it is reliable or not. I think it wasn't a thing back then (would need to check). But agreed w.r.t. scope.